Ckedito5 patch for table cell resize

sharmasahil opened the pull request:
### Jira

[SD-358] - Control WYSIWYG and data table column widths in ckeditor

### Fix

1. Removed local patch as it got issues with retaining the styles while we edit the node again, we are already using Limit allowed html filter which is causing it.
2. Found this sandbox module https://git.drupalcode.org/sandbox/s_leu-3413350/-/tree/1.0.x which solves this issue and allows table inline styles to be saved correctly. It provides a new filter in text format 'Resize table columns' enabling it sorts col resize and retain styles on edit as well.
3. Updated basic enhancer to provide table and col inline styles to FE.


[SD-358]: https://digital-vic.atlassian.net/browse/SD-358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

vincent-gao reviewed (changes requested):
hey @sharmasahil 
do you need an update hook to apply the changes across all sites?
Also, could you clarify the need for a local patch?

sharmasahil commented:
> hey @sharmasahil do you need an update hook to apply the changes across all sites? Also, could you clarify the need for a local patch?

Thanks for the review @vincent-gao, I have updated the PR with comments and also removed local patch, put install hook too.

vincent-gao commented on modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php:
```diff
@@ -82,4 +83,73 @@ public function replaceUnicodeWhitespace(array $data) {
+
+      // Extract only the inner HTML of the <body> tag, avoiding extra tags.
+      $processedBodyContent = '';
+      foreach ($processedDom->getElementsByTagName('body')->item(0)->childNodes as $node) {
```
can this be simplified by using `DOMDocument::saveHTML` directly on the <body> element.
e.g.  
```php
$processedBodyContent = $processedDom->saveHTML($processedDom->getElementsByTagName('body')->item(0));
```

vincent-gao commented on modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php:
```diff
@@ -82,4 +83,73 @@ public function replaceUnicodeWhitespace(array $data) {
+      libxml_use_internal_errors(true);
+      $valueDom->loadHTML($data['value']);
+      $processedDom->loadHTML($data['processed']);
+      libxml_clear_errors();
```
👍

vincent-gao commented on modules/tide_ckeditor/modules/ckeditor_tablecol_resize/js/ckeditor5_plugins/tableColResize/src/tablecolresizeediting.js:
```diff
@@ -0,0 +1,105 @@
+      dispatcher.on( `element:col`, ( evt, data, conversionApi ) => {
+        const { schema, writer } = conversionApi;
+        const colWidth = data.viewItem.getAttribute( dataAttribute );
+        console.log('col upcast', colWidth);
```
the console.log statement should be removed or replaced with a proper logging mechanism if needed for debugging.

vincent-gao reviewed (approved):
hey @sharmasahil 
Just a few minor things, and it's perfectly fine if you don't want to make any changes.
